src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -223,20 +223,50 @@ protected override CompletionItemRules GetCompletionItemRules(ImmutableArray<(IS
                     item = SymbolCompletionItem.AddShouldProvideParenthesisCompletion(item);
                 }
             }
-            else if (symbol.IsKind(SymbolKind.NamedType) || symbol is IAliasSymbol aliasSymbol && aliasSymbol.Target.IsType)
+            else if (shouldProvideParenthesisCompletion(symbol, out var includeDotCommitCharacter))
             {
                 // If this is a type symbol/alias symbol, also consider appending parenthesis when later, it is committed by using special characters,
                 // and the type is used as constructor
+                // We add a flag to disallow automatic parenthesis if the type has a nested type AND the trigger character is '.', in which case the user may want
+                // to access the nested type. See https://github.com/dotnet/roslyn/issues/51629
                 if (context.IsObjectCreationTypeContext)
-                    item = SymbolCompletionItem.AddShouldProvideParenthesisCompletion(item);
+                    item = SymbolCompletionItem.AddShouldProvideParenthesisCompletion(item, includeDotCommitCharacter);
```

Comment:
🙂 You might also add this logic into ObjectCreationCompletionProvider

Reply:
I think I need to understand more the difference in purpose between the two completion providers. We should also make sure the type import completion works well too.

Reply:
> We should also make sure the type import completion works well too.

Good point. I don't think we add "()" for those unimported types now, right? @Cosifne

Reply:
@genlu https://github.com/dotnet/roslyn/blob/b8476018c13f75b8f3bbda9884a153e67a0f260b/src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractImportCompletionProvider.cs#L113

Reply:
> I think I need to understand more the difference in purpose between the two completion providers

`ObjectCreationCompletionProvider` would provide inferred types at current location after "new" with high priority (for pre-selection). Those items would overwritten items for same typed provided by `SymbolCompletionProvider` (those generally have relatively lower priority), since `SymbolCompletionProvider` is ordered before `ObjectCreationCompletionProvider`.

Reply:
@genlu But if `ObjectCreationCompletionProvider` overrides completion items when we are in a location after `new`, why would `SymbolCompletionProvider` ever completes `()`?

Reply:
it only happens for inferred types. For things like `var x = new $$`, it's all `SymbolCompletionProvider`

Reply:
@genlu Ah, so `var x = new $$` is `ObjectCreationCompletionProvider` but `C x = new $$` is `SymbolCompletionProvider` right?

Reply:
I believe it's the opposite, `var x = new $$` is `SymbolCompletionProvider` and `C x = new $$` is `ObjectCreationCompletionProvider`. :)

Reply:
> > We should also make sure the type import completion works well too.
> 
> Good point. I don't think we add "()" for those unimported types now, right? @Cosifne

I don't think we need to do that for unimported types